client/src/components/revealComplete.js

```diff
@@ -21,47 +26,72 @@ const ButtonContainer = styled.div`
 const getRevealedWaifus = async (WETStore) => {
   const oldWaifuIds = WETStore.ownedItems.map(item => item.id);
   await WETStore.syncOwnedItems();
+  const newWaifuIds = WETStore.ownedItems.map(item => item.id);
   const newWaifus = WETStore.ownedItems;
   return newWaifus.filter(waifu => !oldWaifuIds.includes(waifu.id));
+  return newWaifuIds.filter(newId => !oldWaifuIds.includes(newId));
 }
 
 const RevealComplete = observer(({ show }) => {
   const rootStore = useContext(RootStoreContext);
   const { WETStore } = rootStore;
-  const [revealedWaifus, setRevealedWaifus] = useState([])
+  const [revealedWaifuIds, setRevealedWaifuIds] = useState([]);
+  const [loading, setLoading] = useState(false);
   
-  useEffect(() => {
-    if (show) {
-      getRevealedWaifus(WETStore).then((value) => {
-        setRevealedWaifus(value);
-      })
-    }
-  })
+  useEffect(
+    () => {
+      if (show) {
+        setLoading(true);
+        getRevealedWaifus(WETStore)
+        .then((value) => {
+          setRevealedWaifuIds(value);
```

Comment:
I'm assuming you meant `value.id` here!

Reply:
Ahh, I had some `console.logs` around lines 28-30 which didn't remove the old waifus code. Is fixed now: https://github.com/chase-manning/waifusion-dungeon/pull/63/commits/2d736059896b5a3433311fc5cb720c0aba8a1205

Reply:
Ahh, I had some uncommited console.logs around lines 28-30 which didn't remove the old waifu array code. Is fixed now: 2d73605